static/js/lib/base64.js

```diff
@@ -0,0 +1,194 @@
+/*
```

Comment:
Where did you get that library from? Worth a note in the comment imho.

I just read https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem and there appear to be various other changes that could be simpler though I'm fine with using a library as long as we document where to go to for updates and bugfixes.